src/coreclr/jit/importercalls.cpp

```diff
@@ -7754,10 +7758,16 @@ void Compiler::impCheckCanInline(GenTreeCall*           call,
                 return;
             }
 
+            CORINFO_CONTEXT_HANDLE exactContext = pParam->exactContextHnd;
+            if (pParam->call->IsGuardedDevirtualizationCandidate())
+            {
+                exactContext = pParam->call->GetGDVCandidateInfo(pParam->candidateIndex)->exactContextHnd;
+            }
+
```

Comment:
Maybe move this into `impMarkInlineCandidateHelper`? That's where the other handling for candidates happen, it can just update the exact context handle being passed through.

Reply:
it will override the context for a candidate completely, can't do it - explained in https://github.com/dotnet/runtime/pull/87847#issuecomment-1617727629

Reply:
Is it a bug in the EE? Or in the JIT? What would the exact context returned by the EE be if we were inlining the same call in the normal way? Presumably that context is the one we want/expect `resolveVirtualCall` to return. Maybe @davidwrighton has an idea about what we should expect to happen for these kinds of devirtualizations.

Reply:
I still would like to understand this a bit better -- is it possible to add a test I can look at?

Reply:
@EgorBo It still doesn't seem like any test was added here.

Reply:
Oops, missed that. https://github.com/dotnet/runtime/pull/89493